llvm/utils/lit/lit/reports.py

```diff
@@ -14,11 +16,43 @@ def by_suite_and_test_path(test):
     return (test.suite.name, id(test.suite), test.path_in_suite)
 
 
-class JsonReport(object):
+class Report(object):
     def __init__(self, output_file):
         self.output_file = output_file
+        # Set by the option parser later.
+        self.use_unique_output_file_name = False
 
     def write_results(self, tests, elapsed):
+        if self.use_unique_output_file_name:
+            file = None
+            filepath = self.output_file
+            attempt = 0
+            while file is None:
```

Comment:
Rather than looping and incrementing a counter, have you considered either using a UUID as a name suffix or `tempfile.mkstemp`? The latter should provide the functionality you need, unless you specifically want the incrementing behaviour, of course.

Reply:
I don't need the incrementing behaviour. If we want to know the order testing took place in, we should look for and/or add timestamps into the test results themselves.

I changed to using mkstemp, much cleaner now!